allennlp/nn/initializers.py

```diff
@@ -147,17 +148,15 @@ def __call__(self, module: torch.nn.Module) -> None:
         """
         logger.info("Initializing parameters")
         unused_regexes = set([initializer[0] for initializer in self._initializers])
-        uninitialized_parameters = set()
+        uninitialized_parameters = set([name for name, _ in module.named_parameters()])
         # Store which initialisers were applied to which parameters.
-        for name, parameter in module.named_parameters():
-            for initializer_regex, initializer in self._initializers:
-                if re.search(initializer_regex, name):
+        for initializer_regex, initializer in self._initializers:
+            for name, parameter in module.named_parameters():
+                if name in uninitialized_parameters and re.search(initializer_regex, name):
```

Comment:
I'm not sure this is desirable - it could lead to some pretty hard to find bugs. None of the other parameters passed in the JSON respect any kind of ordering, making this unintuitive. Is there a reason you can't just use more specific regexes?